lib/msal-common/src/network/ThrottlingUtils.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { NetworkResponse } from "./NetworkManager";
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { Constants, HeaderNames } from "../utils/Constants";
+import { CacheManager } from "../cache/CacheManager";
+import { StringUtils } from "../utils/StringUtils";
+import { ServerError } from "../error/ServerError";
+import { RequestThumbprint } from "./RequestThumbprint";
+import { RequestThumbprintValue } from "./RequestThumbprintValue";
+
+export class ThrottlingUtils {
+
+    /**
+     * Prepares a RequestThumbprint to be stored as a key.
+     * @param thumbprint
+     */
+    static generateThrottlingStorageKey(thumbprint: RequestThumbprint): string {
+        return `${Constants.THROTTLE_PREFIX}.${JSON.stringify(thumbprint)}`;
+    }
+
+    /**
+     * Performs necessary throttling checks before a network request.
+     * @param cacheManager
+     * @param thumbprint 
+     */
+    static preProcess(cacheManager: CacheManager, thumbprint: RequestThumbprint): void {
+        const key = ThrottlingUtils.generateThrottlingStorageKey(thumbprint);
+        const storageValue = cacheManager.getItem(key) as string;
```

Comment:
What cacheSchema does this fall under? Temporary? Throttling?

Reply:
Thanks for this, had forgotten about the cache schema stuff, it's temporary. Changing